### ISSUE TITLE
`r-check` in `rchk` container does not run

### ISSUE THREAD

**Issue by chainsawriot** [r-hub/containers#67]
I followed the instruction here:

https://r-hub.github.io/containers/local.html

To run the `rchk` container locally and I think the `do-check` function in the`r-check` script inside the image is incomplete.

https://github.com/r-hub/containers/blob/0138fec2416feaf287b2026f9c7679323ce2d319/containers/rchk/r-check#L6-L18

My bash-fu is not the best. But this at least runs in an interactive session:

```sh
do_check () {
    pkg=${1}
    echo Checking "$pkg"
    (
        cd `dirname $pkg`
        R -q -e "pak::pkg_install('deps::$pkg', dependencies = list(direct = 'LinkingTo', indirect = 'hard'))"
        tar xzf $pkg
        topdir="`tar tzf $pkg | head -1`"
        #PACKAGE="`dirname $topdir`"
        #$topdir
        cd $checkdir/$topdir
        /opt/R/devel-rchk/bin/rchk.sh
    )
}
```

In summary

1. pak::pkg_install lacks a closing paren.
2. `topdir` is incorrectly defined
3. I need to take a hackish approach to make the last `cd` work
4. The path to `rchk.sh` is incorrect

**Comment by jameslamb** [comment]
I think this can be closed.

It appears that all of these things were fixed the same day that this issue was opened (https://github.com/r-hub/containers/commit/35ef852efda9f76db642fab66100f9c596d510eb), but just not linked to this issue.

I can see now that the latest version of the image can be used to reproduce CRAN's checks. For example, CRAN currently shows the following for `{data.table}` 1.15.4:

```text
Package data.table version 1.15.4
Package built using 86233/R 4.5.0; x86_64-pc-linux-gnu; 2024-04-01 07:07:33 UTC; unix   
Checked with rchk version fdc068715daa3a256062cc20e0d4a5157dacc9a4 LLVM version 14.0.6
More information at https://github.com/kalibera/cran-checks/blob/master/rchk/PROTECT.md
For rchk in docker image see https://github.com/kalibera/rchk/blob/master/doc/DOCKER.md

Function checkVars
  [UP] allocating function uniq_diff may destroy its unprotected argument (measure_int_or_list <arg 1>), which is later used. data.table/src/fmelt.c:251
  [UP] calling allocating function uniq_diff with a fresh pointer (measure_int_or_list <arg 1>) data.table/src/fmelt.c:251
  [UP] unprotected variable measure_int_or_list while calling allocating function Rf_allocVector data.table/src/fmelt.c:254

Function coerceToRealListR
  [UP] calling allocating function coerceAs(?,V,?) with argument allocated using Rf_ScalarReal data.table/src/frollR.c:18

Function frollapplyR
  [UP] calling allocating function coerceAs(?,V,?) with argument allocated using Rf_ScalarReal data.table/src/frollR.c:260

Function frollfunR
  [UP] calling allocating function coerceAs(?,V,?) with argument allocated using Rf_ScalarReal data.table/src/frollR.c:148

Function gmean
  [UP] calling allocating function coerceAs(?,V,?) with argument allocated using Rf_ScalarReal data.table/src/gsumm.c:594

Function uniq_diff
  [UP] calling allocating function set_diff(V,?) with a fresh pointer (unique_col_numbers <arg 1>) data.table/src/fmelt.c:201
```

([logs link](https://raw.githubusercontent.com/kalibera/cran-checks/master/rchk/results/data.table.out))

I see similar results using the latest images here.

```shell
rm -rf ./packages
mkdir -p ./packages
wget \
    -O ./packages/data.table_1.15.4.tar.gz \
    https://cran.r-project.org/src/contrib/Archive/data.table/data.table_1.15.4.tar.gz

docker run \
    --rm \
    --platform linux/amd64 \
    -v "$(pwd)/packages":/check \
    -it ghcr.io/r-hub/containers/rchk:latest \
    r-check
```

```text
==== rchk bcheck =========================================
ERROR: too many states (abstraction error?) in function strptime_internal
ERROR: too many states (abstraction error?) in function bcEval_loop
ERROR: too many states (abstraction error?) in function StringValue
ERROR: too many states (abstraction error?) in function RunGenCollect
ERROR: too many states (abstraction error?) in function rbindlist
ERROR: too many states (abstraction error?) in function freadMain
ERROR: too many states (abstraction error?) in function rbindlist

Function checkVars
  [UP] allocating function uniq_diff may destroy its unprotected argument (measure_int_or_list <arg 1>), which is later used. /check/data.table/src/fmelt.c:251
  [UP] calling allocating function uniq_diff with a fresh pointer (measure_int_or_list <arg 1>) /check/data.table/src/fmelt.c:251
  [UP] unprotected variable measure_int_or_list while calling allocating function Rf_allocVector /check/data.table/src/fmelt.c:254

Function coerceToRealListR
  [UP] calling allocating function coerceAs(?,V,?) with argument allocated using Rf_ScalarReal /check/data.table/src/frollR.c:18

Function frollapplyR
  [UP] calling allocating function coerceAs(?,V,?) with argument allocated using Rf_ScalarReal /check/data.table/src/frollR.c:260

Function frollfunR
  [UP] calling allocating function coerceAs(?,V,?) with argument allocated using Rf_ScalarReal /check/data.table/src/frollR.c:148

Function gmean
  [UP] calling allocating function coerceAs(?,V,?) with argument allocated using Rf_ScalarReal /check/data.table/src/gsumm.c:594

Function rbindlist
  [UP] protect stack is too deep, unprotecting all variables, results will be incomplete
  [UP] protect stack is too deep, unprotecting all variables, results will be incomplete
  [UP] protect stack is too deep, unprotecting all variables, results will be incomplete
  [UP] unsupported form of unprotect, unprotecting all variables, results will be incomplete /check/data.table/src/rbindlist.c:532

Function shift
  [UP] protect stack is too deep, unprotecting all variables, results will be incomplete
  [UP] protect stack is too deep, unprotecting all variables, results will be incomplete
  [UP] unsupported form of unprotect, unprotecting all variables, results will be incomplete /check/data.table/src/shift.c:174

Function uniq_diff
  [UP] calling allocating function set_diff(V,?) with a fresh pointer (unique_col_numbers <arg 1>) /check/data.table/src/fmelt.c:201
Analyzed 298 functions, traversed 4261426 states.
------------------------------------------------------
```

Suspect the extra stuff I see might just be due to whatever the specifics are of this statement in the docs CRAN links to ([docs link](https://raw.githubusercontent.com/kalibera/cran-checks/master/rchk/README.txt)):

> *The results ... are further filtered in order to reduce the false alarms rate*

<details><summary>output of 'docker inspect' (click me)</summary>

```json
[
    {
        "Id": "sha256:6a9be60319701cc372d97dcefde21b8e866b0fdcb68d10891f962308dec7428d",
        "RepoTags": [
            "ghcr.io/r-hub/containers/rchk:latest"
        ],
        "RepoDigests": [
            "ghcr.io/r-hub/containers/rchk@sha256:5a05030db78164b698ae05025ba383e4129fc9fb53d7aa6b18ebaee9b8aa9c68"
        ],
        "Parent": "",
        "Comment": "buildkit.dockerfile.v0",
        "Created": "2024-10-27T06:17:51.937370329Z",
        "Container": "",
        "ContainerConfig": {
            "Hostname": "",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": null,
            "Cmd": null,
            "Image": "",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": null
        },
        "DockerVersion": "",
        "Author": "",
        "Config": {
            "Hostname": "",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "PATH=/opt/R/devel-rchk/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "PKG_SYSREQS=true",
                "R_PKG_SYSREQS2=true",
                "LC_COLLATE=C",
                "LANG=en_GB.UTF-8",
                "RHUB_ACTIONS_DEPS_DEPENDENCIES=list(direct = \"LinkingTo\", indirect = \"hard\")",
                "RHUB_ACTIONS_CHECK_SCRIPT=rchk.sh"
            ],
            "Cmd": [
                "/bin/bash"
            ],
            "Image": "",
            "Volumes": null,
            "WorkingDir": "",
            "Entrypoint": null,
            "OnBuild": null,
            "Labels": {
                "org.opencontainers.image.ref.name": "ubuntu",
                "org.opencontainers.image.version": "22.04"
            }
        },
        "Architecture": "amd64",
        "Os": "linux",
        "Size": 1558562394,
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/d393ef7a8ee6b016adbcec489f3f78583e8d64bbda627ad907c035bc4b999bea/diff:/var/lib/docker/overlay2/a6e01ee8fc03cecdb6f4b9e26515d4d4d3d562ac39e229f3379e8e85b4646897/diff:/var/lib/docker/overlay2/420567ca3f2376895fd313428bdecae491aad79d2ade2a63ff1044f997330c84/diff:/var/lib/docker/overlay2/1447359a67ad21844ceed9955b3c60f91900117e3e55a4c4a9fce9fd6fd6b303/diff:/var/lib/docker/overlay2/81cdef1d5fc0a75a939d7fa8bb364a3eeb1c937a75a94ac782a6ec014e42657d/diff:/var/lib/docker/overlay2/845ba71cc895148d2fa27afd2cad373df35f817b7bf5f7a38e061adb8867a35a/diff:/var/lib/docker/overlay2/3bdc76a48c79c682be878ed88905f92d7fad4f5b02870e255f05317b55931b1f/diff:/var/lib/docker/overlay2/ec50678f17fb442d64e7a6f3f09b7870073a2e4a659d2ed24b6adbe326f739db/diff",
                "MergedDir": "/var/lib/docker/overlay2/070265bab8704cc3ca0593f270775067fb867984adcc94e22710591b1418b706/merged",
                "UpperDir": "/var/lib/docker/overlay2/070265bab8704cc3ca0593f270775067fb867984adcc94e22710591b1418b706/diff",
                "WorkDir": "/var/lib/docker/overlay2/070265bab8704cc3ca0593f270775067fb867984adcc94e22710591b1418b706/work"
            },
            "Name": "overlay2"
        },
        "RootFS": {
            "Type": "layers",
            "Layers": [
                "sha256:2573e0d8158209ed54ab25c87bcdcb00bd3d2539246960a3d592a1c599d70465",
                "sha256:05e3f5c24f4020229f72d66f375d01921c78f6357f35b81829b73b698b533620",
                "sha256:b4734d2513b0f9b54d32bb9417d51bd028b49b25e85694e47f78ac80342fdb6d",
                "sha256:353977193156ab320892fa7416b8fe4ce64f9f1b75cc7dfa3da251b368a72454",
                "sha256:db5b69321109a9918423e58e692bd16b975e5527e579d6b70099ea1d1c7590b9",
                "sha256:3bcfdcdce81e87ef4ff937806b6a84e766f5bc945afd300a8036d0aba78a1df6",
                "sha256:016d1ef2534513af9864c4df62c292602fe9abe55ced6c346335743c5b6e8b68",
                "sha256:4aecf9b264dc41100b0e3ef70ac1469420c301f8b08573ea0ea01cd0a4d4c8af",
                "sha256:0e3fb1d27335d05550379932005c11b619a14ad78b65c62b01b6826b96729929"
            ]
        },
        "Metadata": {
            "LastTagTime": "0001-01-01T00:00:00Z"
        }
    }
]
```

</details>

**Comment by gaborcsardi** [comment]
Indeed, this was fixed a long time ago, I probably forgot to close it after testing.
